hera_sim/simulate.py

```diff
@@ -61,39 +76,61 @@ def __init__(self, data=None, defaults_config=None, **kwargs):
         self._initialize_data(data, **kwargs)
 
     @property
-    def antpos(self):
-        # TODO: docstring
-        """
-        """
+    def antpos(self) -> Dict[int, Tuple[float, float, float]]:
+        """Dictionary of antenna numbers mapped to ENU positions."""
         antpos, ants = self.data.get_ENU_antpos(pick_data_ants=True)
         return dict(zip(ants, antpos))
 
     @property
-    def lsts(self):
-        # TODO: docstring
+    def lsts(self) -> np.ndarray:
+        """Unique LSTs in the data array."""
         return np.unique(self.data.lst_array)
 
     @property
-    def freqs(self):
+    def freqs(self) -> np.ndarray:
         """Frequencies in GHz."""
         return np.unique(self.data.freq_array) / 1e9
 
     @property
-    def times(self):
-        """Return unique simulation times."""
+    def times(self) -> np.ndarray:
+        """Unique simulation JDs."""
         return np.unique(self.data.time_array)
 
     def apply_defaults(self, config, refresh=True):
-        # TODO: docstring
-        """
-        """
+        """Apply a given set of defaults."""
         # actually apply the default settings
+        # TODO: SGM: why is this a method?
         defaults.set(config, refresh=refresh)
 
-    def add(self, component, **kwargs):
-        # TODO: docstring
-        """
-        """
+    def add(
+        self, component: Union[Type[SimulationComponent], str], **kwargs
+    ) -> Optional[UVData]:
+        """Add a particular simulation component to the simulated data.
+
+        Parameters
+        ----------
+        component
+            Either a string name (or alias) of a component model to add,
+            or its actual class.
+
+        Other Parameters
+        ----------------
+        add_vis
+            Whether to add the simulated component visibilities to the
+            the ``data``. Sometimes you might just want to return the
+            particular component without adding it to the data. Default is True.
+        ret_vis
+            Whether to return the simulated component. Useful if ``add_vis=False``,
+            or you just want a reference to this particular component without being
+            mixed with the rest of the simulation.
+
+        Returns
+        -------
+        Optional[UVData]
+            If ``ret_vis=True``, the data simulated from this component.
+        """
+
+        # TODO: why aren't these just in the signature itself?
```

Comment:
They (as well as some of the other optional parameters) are moved to the signature in the branch where I tidy stuff up in the `Simulator` code. I think at the time of writing I didn't know exactly what optional arguments I wanted/needed, so just left the ``kwargs`` as a catch-all. This shouldn't be an issue once I've finished tidying up the code.

Reply:
OK cool. removing my comment :-)